src/main/cheat_code.cpp

```diff
@@ -1775,7 +1773,7 @@ static void holyTrinity()
         NewEffect(10, tempLocation);
     }
     
-    if(HolyTrinity)
+    if(!ShadowMode && !GodMode && !MultiHop)
```

Comment:
We were saying in the discussion that it would be preferable for holytrinity to enable if *any* of the three were disabled, not if all of the three were.

You may also want to play the grow/shrink sound inside of the subclauses.

Reply:
So, the logic must be:
- When ANY is enabled, enable every other disabled.
- When ALL are enabled, disable them all.
```cpp
if(ShadowMode && GodMode && MultiHop)
{
   // Disable everything
}
else
{
    if(!ShadowMode)
    {
        // enable it
    }

    if(!GodMode)
    {
        // enable it
    }

    if(!MultiHop)
    {
        // enable it
    }
}
```

Reply:
Wouldn't it suffice to do

```c++
if(ShadowMode && GodMode && MultiHop)
{
   // Disable everything
}
else
{
   // Enable everything
}
```

?

Or is there an advantage for the additional check that I'm not seeing?

Reply:
I guess, for such an effect, we better to have one single sound, no need to play the sound of each effect trice :eyes: